crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -708,14 +723,18 @@ mod tests {
             venv_sys_prefix
         }
 
+        fn err(self) -> SitePackagesDiscoveryError {
+            let env_path = self.build();
+            let env = PythonEnvironment::new(env_path, self.origin, &self.system);
+            let Err(err) = env else {
+                panic!("Expected an error, but got: {env:?}");
+            };
+            err
+        }
```

Comment:
```suggestion
        #[track_caller]
        fn err(self) -> SitePackagesDiscoveryError {
            PythonEnvironment::new(self.build(), self.origin, &self.system)
                .expect_err("Expected site-packages discovery to fail")
        }
```

(`expect_err()` prints the contents of the `Ok()` variant in the panic message, as well as the string you pass into the function)

Reply:
👍

Reply:
Why `#[track_caller]`?

Reply:
That's nice!